File: redsheet-portal/src/app/database-models/project-post-event-action.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectModule, vProjectPostEventActionRecord } from './'
export class ProjectPostEventAction {
    constructor(projectPostEventAction?: ProjectPostEventAction | vProjectPostEventActionRecord) {
        let properties = ["postEventActionBy", "postEventActionDetail", "postEventActionOn", "projectPostEventActionId", "projectModuleId",];
        for (let property of properties)
            if (projectPostEventAction && projectPostEventAction[property])
                this[property] = projectPostEventAction[property];
    }

    @required()
    @maxLength(50)
    postEventActionBy: string = undefined;

    @required()
    @maxLength(100)
    postEventActionDetail: string = undefined;

    @required()
    postEventActionOn: Date = undefined;

    projectPostEventActionId: number = 0;

    @range(0, 2147483647)
    projectModuleId: number = undefined;
    projectModule: ProjectModule;


}

File: RedSheet.DbEntities/Models/ProjectPostEventAction.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
    [Table("ProjectPostEventActions", Schema = "dbo")]
    [Serializable]
    public partial class ProjectPostEventAction
    {

        [Required]
        [MaxLength(50)]
        public string PostEventActionBy { get; set; }

        [Required]
        [MaxLength(100)]
        public string PostEventActionDetail { get; set; }

        [Required]
        public System.DateTime PostEventActionOn { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int ProjectPostEventActionId { get; set; }

        [RelationshipTableAttribue("ProjectModules", "dbo")]
        //Relationships
        public int ProjectModuleId { get; set; }

        [ForeignKey("ProjectModuleId")]
        public ProjectModule ProjectModule { get; set; }

        public ProjectPostEventAction()
        {
        }

    }
}

